src/Containers/Microsoft.NET.Build.Containers/FallbackToHttpMessageHandler.cs

```diff
@@ -80,6 +80,7 @@ private static void FallbackToHttp(HttpRequestMessage request)
     {
         var uriBuilder = new UriBuilder(request.RequestUri!);
         uriBuilder.Scheme = "http";
+        uriBuilder.Port = 80;
```

Comment:
This will explicitly set the port to 80 (i.e. `http://whatever:80`) - is that what's intended, or in that scenario would you want the implied default port (i.e. `http://whatever`)? If the latter, then you can set it to `-1`.

Reply:
Thanks for your review. 

Maybe I should change the port only when the port is 443, as the user's registry may specified the port.

Reply:
I tested, If user specified a http port. the publish command will result an error:
```
Uploading layer 'sha256:28d4f52c4a876bde6af159c1e53d159d94edfab99bfdc6eda8b53be82ee29eb1' to 'dockerhub.shengguanda.com:80'
C:\Program Files\dotnet\sdk\8.0.401\Containers\build\Microsoft.NET.Build.Containers.targets(242,5): error CONTAINER1013: Failed to push to the output registry: The SSL connection could not be established, see inner excepti
on. 
```

So there is no way to specify a http port. It is another issue.

Reply:
I think that we should defer to the user's registry configuration here:

* if the registry url had a port, then we should use it
* if the registry url did not have a port, then we should set `-1` here, because per [the docs](https://learn.microsoft.com/en-us/dotnet/api/system.uribuilder.port?view=net-8.0#system-uribuilder-port)

> If the [Port](https://learn.microsoft.com/en-us/dotnet/api/system.uri.port?view=net-8.0#system-uri-port) property is set to a value of -1, this indicates that the default port value for the protocol scheme will be used to connect to the host.

So the 'default' value of the `http` protocol would be used in this case.